From f25965f8b16e6605c4486235961212c947d853f1 Mon Sep 17 00:00:00 2001
From: Zheng Zengkai <zhengzengkai@huawei.com>
Date: Thu, 14 Jul 2022 21:57:22 +0800
Subject: [PATCH] ipv6: fix kabi for mc_forwarding in struct ipv6_devconf
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: f25965f8b16e6605c4486235961212c947d853f1
Modified-by-SEL: No


hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I5GL1Z
CVE: NA

---------------------------------------

Making mc_forwarding atomic breaks the KABI of struct ipv6_devconf.
This patch uses KABI_REPLACE to fix it.

Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Reviewed-by: Wei Li <liwei391@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/ipv6.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/ipv6.h b/include/linux/ipv6.h
index f4e1790a5eed..210402b26a20 100644
--- a/include/linux/ipv6.h
+++ b/include/linux/ipv6.h
@@ -51,7 +51,7 @@ struct ipv6_devconf {
 	__s32		use_optimistic;
 #endif
 #ifdef CONFIG_IPV6_MROUTE
-	atomic_t	mc_forwarding;
+	KABI_REPLACE(__s32 mc_forwarding, atomic_t mc_forwarding)
 #endif
 	__s32		disable_ipv6;
 	__s32		drop_unicast_in_l2_multicast;
-- 
2.34.1

